java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -710,6 +722,23 @@ public Description matchBinary(BinaryTree tree, VisitorState state) {
     return Description.NO_MATCH;
   }
 
+  /**
+   * This method checks if the input (binary) expression tree is equivalent to the final result.
+   * Note that to avoid incorrect removal of parenthesis, we check if the expression is again not a
+   * binary expression.
+   *
+   * @param tree expression tree
+   * @param state visitor state
+   * @param remaining expression after removal
+   * @return true if parenthesis are redundant
+   */
+  private boolean checkLoneCondition(ExpressionTree tree, VisitorState state, String remaining) {
+    return (state.getSourceForNode(tree).equals(remaining)
+        && (!tree.getKind().equals(Kind.CONDITIONAL_OR)
+            && !tree.getKind().equals(Kind.CONDITIONAL_AND))
+        && isTreated);
```

Comment:
why is there a `isTreated` here? This code should not be dependent on the treatment value.

Reply:
to check if the complete expression boils down to a single identifier expression there is a dependency on the treatment value also. This is because it totally depends on which branch one is selecting. This part is somewhat similar to `evalExpr` method call where we check the branch path.

Based on the branch path taken, the child code deletion happens which in turn affects if the parent node is a lone identifier contained in parenthesis.

Reply:
@mkr-plse I do believe that moving this treatment check out of this method and placing this at the top level does makes sense